core/src/banking_stage/worker.rs

```diff
@@ -0,0 +1,626 @@
+        leader_bank_notifier: Arc<LeaderBankNotifier>,
+    ) -> Self {
+        Self {
+            consume_receiver,
+            consumer,
+            consumed_sender,
+            forward_receiver,
+            forward_option,
+            forwarder,
+            forwarded_sender,
+            leader_bank_notifier,
+        }
+    }
+
+    pub fn run(self) -> Result<(), WorkerError> {
+        loop {
+            select! {
+                recv(self.consume_receiver) -> consume_work => {
+                    self.consume_loop(consume_work?)?;
+                },
+                recv(self.forward_receiver) -> forward_work => {
+                    self.forward_loop(forward_work?)?;
+                },
+            }
+        }
+    }
+
+    fn consume_loop(&self, consume_work: ConsumeWork) -> Result<(), WorkerError> {
+        let Some(mut bank) = self.get_consume_bank() else {
+            return self.retry_drain(consume_work);
+        };
+
+        for work in std::iter::once(consume_work).chain(self.consume_receiver.try_iter()) {
+            if bank.is_complete() {
+                if let Some(new_bank) = self.get_consume_bank() {
+                    bank = new_bank;
+                } else {
+                    return self.retry_drain(work);
+                }
+            }
+            self.consume(&bank, work)?;
+        }
+
+        Ok(())
+    }
+
+    /// Consume a single batch.
+    fn consume(&self, bank: &Arc<Bank>, mut consume_work: ConsumeWork) -> Result<(), WorkerError> {
+        let summary = self.consumer.process_and_record_aged_transactions(
+            bank,
+            &consume_work.transactions,
+            &mut consume_work.max_age_slots,
+        );
+
+        self.consumed_sender.send(FinishedConsumeWork {
+            work: consume_work,
+            retryable_indexes: summary
+                .execute_and_commit_transactions_output
+                .retryable_transaction_indexes,
+        })?;
+        Ok(())
+    }
+
+    /// Try to get a bank for consuming.
+    fn get_consume_bank(&self) -> Option<Arc<Bank>> {
+        self.leader_bank_notifier
+            .get_or_wait_for_in_progress(Duration::from_millis(50))
+            .upgrade()
+    }
+
+    /// Retry current batche and all outstanding batches.
+    fn retry_drain(&self, work: ConsumeWork) -> Result<(), WorkerError> {
+        for work in std::iter::once(work).chain(self.consume_receiver.try_iter()) {
+            self.retry(work)?;
+        }
+        Ok(())
+    }
+
+    /// Send transactions back to scheduler as retryable.
+    fn retry(&self, work: ConsumeWork) -> Result<(), WorkerError> {
+        let retryable_indexes = (0..work.transactions.len()).collect();
+        self.consumed_sender.send(FinishedConsumeWork {
+            work,
+            retryable_indexes,
+        })?;
+        Ok(())
+    }
+
+    fn forward_loop(&self, forward_work: ForwardWork) -> Result<(), WorkerError> {
```

Comment:
nit: `work: Forward` to be consistent with `failed_forward_drain`?

Reply:
consistently renamed everything to simply `work`